src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -1610,6 +1610,82 @@ bool SQLStorage::loadEcuReportCounter(std::vector<std::pair<Uptane::EcuSerial, i
   return true;
 }
 
+void SQLStorage::saveReportEvent(const Json::Value& json_value) {
+  std::string json_string = Utils::jsonToCanonicalStr(json_value);
+  SQLite3Guard db = dbConnection();
+  int64_t id = 0;
+  {
+    auto statement = db.prepareStatement("SELECT MAX(id) FROM report_events;");
```

Comment:
The two should be in a transaction.

Or better, it can be done in one step:

```
db.prepareStatement<std::string>("INSERT INTO report_events SELECT MAX(id) + 1, ? FROM report_events", json_string);
```

Reply:
Thank you, modified.

Reply:
I don't see the changes yet?

Reply:
I made a mistake in the last push. It should be updated now.